cmd/soroban-rpc/lib/xdr2json/src/lib.rs

```diff
@@ -50,12 +50,18 @@ pub unsafe extern "C" fn xdr_to_json(
 ) -> *mut ConversionResult {
     let result = catch_json_to_xdr_panic(Box::new(move || {
         let type_str = unsafe { from_c_string(typename) };
-        let the_type = xdr::TypeVariant::from_str(&type_str).unwrap();
+        let the_type = match xdr::TypeVariant::from_str(&type_str) {
+            Ok(t) => t,
+            Err(e) => panic!("couldn't match type {type_str}: {}", e),
+        };
 
         let xdr_bytearray = unsafe { from_c_xdr(xdr) };
         let mut buffer = xdr::Limited::new(xdr_bytearray.as_slice(), DEFAULT_XDR_RW_LIMITS.clone());
 
-        let t = xdr::Type::read_xdr_to_end(the_type, &mut buffer).unwrap();
+        let t = match xdr::Type::read_xdr_to_end(the_type, &mut buffer) {
+            Ok(t) => t,
+            Err(e) => panic!("couldn't read {type_str}: {}", e),
```

Comment:
I dont understand rust that much so wondering that how we use `panic` here is usually encouraged in rust world?

Reply:
No worries, I had to figure this out myself for this.

Since we're "crossing a language boundary" through C code, we need to make sure that errors in the Rust portion are handled correctly. There are two ways to do this: (a) make sure your code never panics, or (b) ["unwind"](https://doc.rust-lang.org/std/panic/fn.catch_unwind.html) all panics. 

I initially did (a) and returned a JSONified `{ "error": "<message>" }` field, but I scrapped it because it meant that the client-side would have to decode and inspect the JSON which meant an extra round of unmarshalling, whereas how it works today is just shoving the bytes returned from Rust directly into the HTTP response. That meant I had to do (b) which is already what we're doing for preflight so I adapted that code accordingly.

You can see that this whole method is wrapped in a call to `catch_json_to_xdr_panic`, which basically says "run this function and catch any panics," formatting the panic message into a returned error string (hence the need for a `ConversionResult` struct rather than just returning the JSON bytes).